rust/xaynet-server/src/storage/redis.rs

```diff
@@ -0,0 +1,644 @@
+    ///
+    /// Returns [`DeleteSumParticipant::Ok`] if field was deleted or
+    /// [`DeleteSumParticipant::DoesNotExist`] if field does not exist.
+    pub async fn remove_sum_dict_entry(
+        mut self,
+        pk: &SumParticipantPublicKey,
+    ) -> RedisResult<DeleteSumParticipant> {
+        debug!(
+            "remove sum dictionary entry for sum participant with pk {:?}",
+            pk
+        );
+        // https://redis.io/commands/hdel
+        // > Return value
+        //   Integer reply: the number of fields that were removed from the hash,
+        //   not including specified but non existing fields.
+        self.connection
+            .hdel("sum_dict", PublicSigningKeyWrite::from(pk))
+            .await
+    }
+
+    /// Retrieves the length of the [`SumDict`].
+    pub async fn get_sum_dict_len(mut self) -> RedisResult<u64> {
+        debug!("get length of sum dictionary");
+        // https://redis.io/commands/hlen
+        // > Return value
+        //   Integer reply: number of fields in the hash, or 0 when key does not exist.
+        self.connection.hlen("sum_dict").await
+    }
+
+    /// Retrieves the [`SumParticipantPublicKey`] of the [`SumDict`] or an empty list when the
+    /// [`SumDict`] does not exist.
+    pub async fn get_sum_pks(mut self) -> RedisResult<HashSet<SumParticipantPublicKey>> {
+        debug!("get public keys of all sum participants");
+        // https://redis.io/commands/hkeys
+        // > Return value:
+        //   Array reply: list of fields in the hash, or an empty list when key does not exist.
+        let result: HashSet<PublicSigningKeyRead> = self.connection.hkeys("sum_dict").await?;
+        let sum_pks = result.into_iter().map(|pk| pk.into()).collect();
+
+        Ok(sum_pks)
+    }
+
+    /// Retrieves the [`SeedDict`] entry for the given ['SumParticipantPublicKey'] or an empty map
+    /// when a [`SeedDict`] entry does not exist.
+    pub async fn get_seed_dict_for_sum_pk(
+        mut self,
+        sum_pk: &SumParticipantPublicKey,
+    ) -> RedisResult<HashMap<UpdateParticipantPublicKey, EncryptedMaskSeed>> {
+        debug!(
+            "get seed dictionary for sum participant with pk {:?}",
+            sum_pk
+        );
+        // https://redis.io/commands/hgetall
+        // > Return value
+        //   Array reply: list of fields and their values stored in the hash, or an empty
+        //   list when key does not exist.
+        let result: Vec<(PublicSigningKeyRead, EncryptedMaskSeedRead)> = self
+            .connection
+            .hgetall(PublicSigningKeyWrite::from(sum_pk))
+            .await?;
+        let seed_dict = result
+            .into_iter()
+            .map(|(pk, seed)| (pk.into(), seed.into()))
+            .collect();
+
+        Ok(seed_dict)
+    }
+
+    /// Retrieves the [`SeedDict`] or an empty [`SeedDict`] when the [`SumDict`] does not exist.
+    pub async fn get_seed_dict(mut self) -> RedisResult<SeedDict> {
```

Comment:
Do we ever need to retrieve the full seed dict? I thought we only needed parts of it for each sum participant.

Reply:
Unfortunately yes, we broadcast the seed dict at the end of the update phase 
https://github.com/xaynetwork/xaynet/pull/519/files#diff-b15272399ff491c84d93d4ca104eaab1R77

Reply:
Oh ok. With these changes though we shouldn't need to do so anymore. `xaynet_server::services::fetchers::SeedDictService` should actually talk to redis directly. We should just send an event saying that a new dict is available. Same goes for the sum dictionary. But better address that in a follow-up PR.